frontend/src/views/SecretMainPage/components/ActionBar/CreateDynamicSecretForm/AzureEntraIdInputForm.tsx

```diff
@@ -0,0 +1,240 @@
+        // a day
+        if (valMs > 24 * 60 * 60 * 1000)
+            ctx.addIssue({ code: z.ZodIssueCode.custom, message: "TTL must be less than a day" });
+    }),
+    maxTTL: z
+        .string()
+        .optional()
+        .superRefine((val, ctx) => {
+            if (!val) return;
+            const valMs = ms(val);
+            if (valMs < 60 * 1000)
+                ctx.addIssue({ code: z.ZodIssueCode.custom, message: "TTL must be a greater than 1min" });
+            // a day
+            if (valMs > 24 * 60 * 60 * 1000)
+                ctx.addIssue({ code: z.ZodIssueCode.custom, message: "TTL must be less than a day" });
+        }),
+    name: z.string().min(1).refine((val) => val.toLowerCase() === val, "Must be lowercase")
+});
+type TForm = z.infer<typeof formSchema>;
+
+type Props = {
+    onCompleted: () => void;
+    onCancel: () => void;
+    secretPath: string;
+    projectSlug: string;
+    environment: string;
+};
+
+type EntraIDUser = {
+    name: string;
+    id: string;
+    email: string;
+};
+
+export const AzureEntraIdInputForm = ({
+    onCompleted,
+    onCancel,
+    environment,
+    secretPath,
+    projectSlug
+}: Props) => {
+    const {
+        control,
+        formState: { isSubmitting },
+        watch,
+        handleSubmit
+    } = useForm<TForm>({
+        resolver: zodResolver(formSchema)
+    });
+    const tenantId = watch("provider.tenantId");
+    const [selectedUsers, setSelectedUsers] = useState<EntraIDUser[]>([]);
+    const { data, isLoading, isFetched, isError, isFetching } = useGetDynamicSecretProviderData({ dataFetchType: "Users", provider: {type: DynamicSecretProviders.AzureEntraId, inputs: { userId : "unused", email: "unused", tenantId }}, enabled: !!tenantId });
+    const mappedUserList = data?.map((user) => ({
+        primaryText: user.name,
+        secondaryText: user.email,
+        id: user.id,
+      }));
+    const handleChange = (selected: { primaryText: string; secondaryText: string; id: string }[]) => {
+        // Map back to EntraIDUser based on the id or other attributes
+        const selectedMappedUsers = data?.filter(user =>
+          selected.some(selectedItem => selectedItem.id === user.id)
+        );
+        if(selectedMappedUsers){
+            setSelectedUsers(selectedMappedUsers);
+        }
+      };
+    const createDynamicSecret = useCreateDynamicSecret();
+
+    const handleCreateDynamicSecret = async ({ name, provider, maxTTL, defaultTTL }: TForm) => {
+        // wait till previous request is finished
+        if (createDynamicSecret.isLoading) return;
+        try {
+            selectedUsers.map(async (user: { id: string, name: string, email: string }) => {
+                await createDynamicSecret.mutateAsync({
+                    provider: { type: DynamicSecretProviders.AzureEntraId, inputs: { userId: user.id, tenantId: provider.tenantId, email: user.email } },
+                    maxTTL,
+                    name: `${name}-${user.name}`,
```

Comment:
Please keep provider specific steps in here and not the other side. That is generation of tenant id